azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_ops_provisioning.py

```diff
@@ -124,7 +124,7 @@ class SendQueryRequestOperation(PipelineOperation):
     This operation is in the group of DPS operations because it is very specific to the DPS client.
     """
 
-    def __init__(self, request_id, operation_id, request_payload, callback):
+    def __init__(self, operation_id, request_payload, registration_result=None, callback=None):
```

Comment:
>callback=None [](start = 80, length = 13)

Please don't make callbacks optional  This is important.  Here and everywhere.

Reply:
+1, it makes dealing with a lot of the difficult connective tissue of the pipeline much easier when there is always a callback.

---
In reply to: [364361479](https://github.com/Azure/azure-iot-sdk-python/pull/432#discussion_r364361479) [](ancestors = 364361479)